warehouse/slave.go

```diff
@@ -189,126 +207,170 @@ func (job *Payload) getDiscardsTable() string {
 	return warehouseutils.ToProviderCase(job.DestinationType, warehouseutils.DiscardsTable)
 }
 
-func (jobRun *JobRun) getLoadFilePath(tableName string) string {
-	job := jobRun.job
+func (jr *JobRun) getLoadFilePath(tableName string) string {
+	job := jr.job
 	randomness := misc.FastUUID().String()
-	return strings.TrimSuffix(jobRun.stagingFilePath, "json.gz") + tableName + fmt.Sprintf(`.%s`, randomness) + fmt.Sprintf(`.%s`, warehouseutils.GetLoadFileFormat(job.LoadFileType))
+	return strings.TrimSuffix(jr.stagingFilePath, "json.gz") + tableName + fmt.Sprintf(`.%s`, randomness) + fmt.Sprintf(`.%s`, warehouseutils.GetLoadFileFormat(job.LoadFileType))
 }
 
 func (job *Payload) getColumnName(columnName string) string {
 	return warehouseutils.ToProviderCase(job.DestinationType, columnName)
 }
 
-type loadFileUploadJob struct {
-	tableName  string
-	outputFile encoding.LoadFileWriter
-}
-
-type loadFileUploadOutput struct {
-	TableName             string
-	Location              string
-	TotalRows             int
-	ContentLength         int64
-	StagingFileID         int64
-	DestinationRevisionID string
-	UseRudderStorage      bool
-}
+// uploadLoadFiles returns the upload output for each file uploaded to object storage
+func (jr *JobRun) uploadLoadFiles(ctx context.Context) ([]loadFileUploadOutput, error) {
+	ctx, cancel := context.WithTimeout(ctx, jr.slaveUploadTimeout)
+	defer cancel()
 
-func (jobRun *JobRun) uploadLoadFilesToObjectStorage(ctx context.Context) ([]loadFileUploadOutput, error) {
-	job := jobRun.job
-	uploader, err := job.getFileManager(job.DestinationConfig, job.UseRudderStorage)
+	uploader, err := jr.job.getFileManager(jr.job.DestinationConfig, jr.job.UseRudderStorage)
 	if err != nil {
-		return []loadFileUploadOutput{}, err
-	}
-	// var loadFileIDs []int64
-	var loadFileUploadOutputs []loadFileUploadOutput
-
-	// take the first staging file id in upload
-	// TODO: support multiple staging files in one upload
-	stagingFileId := jobRun.job.StagingFileID
-
-	loadFileOutputChan := make(chan loadFileUploadOutput, len(jobRun.outputFileWritersMap))
-	loadFileUploadTimer := jobRun.timerStat("load_file_upload_time")
-	uploadJobChan := make(chan *loadFileUploadJob, len(jobRun.outputFileWritersMap))
-	// close chan to avoid memory leak ranging over it
-	defer close(uploadJobChan)
-	uploadErrorChan := make(chan error, numLoadFileUploadWorkers)
-	ctx, cancel := context.WithCancel(ctx)
-	defer cancel()
-	for i := 0; i < numLoadFileUploadWorkers; i++ {
-		go func(ctx context.Context) {
-			for uploadJob := range uploadJobChan {
+		return nil, fmt.Errorf("creating uploader: %w", err)
+	}
+
+	type result struct {
+		uploadOutput loadFileUploadOutput
+		err          error
+	}
+
+	type uploadJob struct {
+		tableName  string
+		outputFile encoding.LoadFileWriter
+	}
```

Comment:
My preference would be to have these outside the `uploadLoadFiles` method, which is already quite long.

Reply:
Moved it outside.